pkg/chains/formats/slsa/extract/extract.go

```diff
@@ -34,12 +34,67 @@ import (
 
 // SubjectDigests returns software artifacts produced from the TaskRun/PipelineRun object
 // in the form of standard subject field of intoto statement.
-// The type hinting fields expected in results help identify the generated software artifacts.
+// The type hinting fields expected in TaskRun results help identify the generated software artifacts in a TaskRun/PipelineRun.
 // Valid type hinting fields must:
 //   - have suffix `IMAGE_URL` & `IMAGE_DIGEST` or `ARTIFACT_URI` & `ARTIFACT_DIGEST` pair.
 //   - the `*_DIGEST` field must be in the format of "<algorithm>:<actual-sha>" where the algorithm must be "sha256" and actual sha must be valid per https://github.com/opencontainers/image-spec/blob/main/descriptor.md#sha-256.
 //   - the `*_URL` or `*_URI` fields cannot be empty.
 func SubjectDigests(ctx context.Context, obj objects.TektonObject) []intoto.Subject {
+	var subjects []intoto.Subject
+
+	switch obj.GetObject().(type) {
+	case *v1beta1.PipelineRun:
```

Comment:
Consider adding new methods to the TektonObject interface to avoid having to treat each different implementation differently. For example, you may want to add a `TaskRuns()` method that returns a list of the TaskRuns with the object.

Reply:
SGTM!

Reply:
Let me open a separate PR to address this. Since this would help us clean up some other parts where we need to access child taskruns ([example](https://github.com/tektoncd/chains/blob/5230749f22aaa1a81305a2bf8dd7616614609d66/pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go#L181-L186)).

Reply:
I'll close this comment and Please review https://github.com/tektoncd/chains/pull/879 instead. Thanks